### ISSUE TITLE
Intermittent failure in /_mozilla/mozilla/promise.html

### ISSUE THREAD

**Issue by jdm** [servo/servo#13384]
```
  ▶ Unexpected subtest result in /_mozilla/mozilla/promise.html:
  │ FAIL [expected PASS] Native promise from async callback can be resolved
  │   → assert_true: expected true got false
  │ 
  └ @http://web-platform.test:8000/_mozilla/mozilla/promise.html:58:11
```

**Comment by jdm** [comment]
This is the `assert_true(end - start > 100);` assertion failing. I'm surprised, since the resolution isn't supposed to occur for 100ms, and it uses the same mechanism internally as regular web timers.

**Comment by jdm** [comment]
I think we can call this fixed by #13385.
